tests/external_botocore/test_get_ai_message_ids.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2010 New Relic, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from mlmodel_bedrock.test_chat_completion import (  # noqa: F401; pylint: disable=W0611
+    exercise_model,
+    is_file_payload,
+    model_id,
+)
+from testing_support.fixtures import (  # noqa: F401; pylint: disable=W0611
+    override_application_settings,
+    reset_core_stats_engine,
+)
+
+from newrelic.api.background_task import background_task
+from newrelic.api.ml_model import get_ai_message_ids
+from newrelic.api.time_trace import current_trace
+from newrelic.api.transaction import add_custom_attribute, current_transaction
+
+_test_bedrock_chat_completion_prompt = "What is 212 degrees Fahrenheit converted to Celsius?"
+
+
+def set_trace_info():
+    txn = current_transaction()
+    if txn:
+        txn._trace_id = "trace-id"
+    trace = current_trace()
+    if trace:
+        trace.guid = "span-id"
+
+
+@reset_core_stats_engine()
+@background_task()
+def test_get_ai_message_ids_when_nr_message_ids_not_set():
+    message_ids = get_ai_message_ids("request-id-1")
+    assert message_ids == []
+
+
+@reset_core_stats_engine()
+def test_get_ai_message_ids_outside_transaction():
+    message_ids = get_ai_message_ids("request-id-1")
+    assert message_ids == []
+
+
+@reset_core_stats_engine()
+def test_get_ai_message_ids_bedrock_chat_completion_in_txn(exercise_model):  # noqa: F811
+    @background_task()
+    def _test():
+        set_trace_info()
+        add_custom_attribute("conversation_id", "my-awesome-id")
+        exercise_model(prompt=_test_bedrock_chat_completion_prompt, temperature=0.7, max_tokens=100)
+        message_ids = [m for m in get_ai_message_ids()]
+
+        for message_id_info in message_ids:
```

Comment:
This one test, tests all the models which is great but the assertions will need to be catered towards the model under tests. I'd suggest creating another expected dict in the _test_chat_completion.py with the expected ai_message_ids for each model. I'd also suggest moving these tests into the test_chat_completion.py as I think it will be easier in the long run since all the fixtures and expected values are in there that are getting re-used here.

There are 3 primary test cases we need to cover here:

In the case where bedrock has a response_id (the Jurassic model), you should be able to make this a hardcoded check as all of these id's will be repeatable within the transaction.

In the case where bedrock does not provide a response_id (the titan model). Maybe make the value None in the expected dictionary and then in cases where it's None in this test's assertion check for presence like you are doing here instead of asserting the value. This is similar to what we do in some of our validators.

The case where there is no conversation_id set (aka conversation_id == "").